hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -0,0 +1,439 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.async.HoodieAsyncService;
+import org.apache.hudi.client.common.HoodieSparkEngineContext;
+import org.apache.hudi.common.config.HoodieMetadataConfig;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.fs.FSUtils;
+import org.apache.hudi.common.model.FileSlice;
+import org.apache.hudi.common.model.HoodieBaseFile;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.view.FileSystemViewManager;
+import org.apache.hudi.common.table.view.HoodieTableFileSystemView;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.exception.HoodieException;
+import org.apache.hudi.exception.HoodieValidationException;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaSparkContext;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+/**
+ * A validator with spark-submit to compare list partitions and list files between metadata table and filesystem
+ * <p>
+ * You can run this validator with the following command and have a try:
+ * ```
+ * spark-submit \
+ *  --class org.apache.hudi.utilities.HoodieMetadataTableValidator \
+ *  --packages org.apache.spark:spark-avro_2.11:2.4.4 \
+ *  --master spark://xxxx:7077 \
+ *  --driver-memory 1g \
+ *  --executor-memory 1g \
+ *  $HUDI_DIR/hudi/packaging/hudi-utilities-bundle/target/hudi-utilities-bundle_2.11-0.11.0-SNAPSHOT.jar \
+ *  --base-path basePath \
+ *  --mode CONTINUOUS
+ * ```
+ *
+ * <p>
+ * You can specify the running mode of the validator through `--mode`.
+ * There are 2 modes of the {@link HoodieMetadataTableValidator}:
+ * - CONTINUOUS : This validator will compare the result of listing partitions/listing files between metadata table and filesystem every 10 minutes(default).
+ * <p>
+ * Example command:
+ * ```
+ * spark-submit \
+ *  --class org.apache.hudi.utilities.HoodieMetadataTableValidator \
+ *  --packages org.apache.spark:spark-avro_2.11:2.4.4 \
+ *  --master spark://xxxx:7077 \
+ *  --driver-memory 1g \
+ *  --executor-memory 1g \
+ *  $HUDI_DIR/hudi/packaging/hudi-utilities-bundle/target/hudi-utilities-bundle_2.11-0.11.0-SNAPSHOT.jar \
+ *  --base-path basePath \
+ *  --mode CONTINUOUS
```

Comment:
i don't see much difference between example 1 and example2. May be you also add --min-validate-interval-seconds to the example. and have just 1 example for continuous and 1 example for once mode.

Reply:
Sure thing. Added. 
PTAL :)